src/server/gc_worker/compaction_filter.rs

```diff
@@ -287,14 +287,15 @@ struct WriteCompactionFilter {
     gc_scheduler: Scheduler<GcTask<RocksEngine>>,
     // A key batch which is going to be sent to the GC worker.
     mvcc_deletions: Vec<Key>,
-    // The count of records covered the current mvcc-deletion mark. The mvcc-deletion
-    // mark will be sent to the GC worker only if `mvcc_deletion_overlaps` is 0. It's
-    // a little optimization to reduce modifications on write CF.
-    mvcc_deletion_overlaps: Option<usize>,
+    // The mvcc-deletion mark will be sent to the GC worker only if `mvcc_del_gc_needed` is
+    // true. It's a little optimization to reduce modifications on write CF.
```

Comment:
The *little optimization* still exist here? Previously, `GcKeys` task is send only when: 1) the latest version before safepoint is `DELETE` and 2) we don't find any older version than the `DELETE` of the key. In case there are multiple versions before safepoint and the latest one is `DELETE`, the `mvc_deletion_overlaps` will be `Some(x)` where `x > 0`, so `handle_bottommost_delete` is skipped in this case. This optimization doesn't seem to be kept after your change. 😕

Reply:
Oh，I kept this logic:  If any older version should be removed in compaction filter, we shouldn't gc_keys for this key